numpy/testing/_private/utils.py

```diff
@@ -404,6 +389,31 @@ def assert_equal(actual, desired, err_msg='', verbose=True):
     except (TypeError, ValueError, NotImplementedError):
         pass
 
+    # Inf/nan/negative zero handling
+    try:
+        isdesnan = gisnan(desired)
+        isactnan = gisnan(actual)
+        if isdesnan and isactnan:
+            return  # both nan, so equal
+
+        # handle signed zero specially for floats
+        array_actual = array(actual)
+        array_desired = array(desired)
+        if (array_actual.dtype.char in 'Mm' or
+                array_desired.dtype.char in 'Mm'):
+            # version 1.18
+            # until this version, gisnan failed for datetime64 and timedelta64.
+            # Now it succeeds but comparison to 0 emits a DeprecationWarning
+            # Avoid that by skipping the next check
+            raise NotImplementedError('cannot compare datetime64 to 0')
```

Comment:
you prefer a bare `raise NotImplementedError()` to the current exception which is explicitly caught on line 414?

Reply:
The comment should be updated, too.  It is not just comparison to 0 that results in the `DeprecationWarning`.

Reply:
Since the exception will be caught a few lines down, is there are need for the message?  A message is fine if you prefer having one, but it should be accurate.  The current message (about comparing to 0) is misleading.